cuda/test/matrix/csr_kernels.cpp

```diff
@@ -219,6 +194,18 @@ TEST_F(Csr, SimpleApplyIsEquivalentToRefWithLoadBalance)
 }
 
 
+TEST_F(Csr, SimpleApplyIsEquivalentToRefWithLoadBalanceUnsorted)
+{
+    set_up_apply_data(std::make_shared<Mtx::load_balance>(cuda));
+    unsort_mtx();
+
+    mtx->apply(y.get(), expected.get());
+    dmtx->apply(dy.get(), dresult.get());
+
+    GKO_ASSERT_MTX_NEAR(dresult, expected, 1e-14);
+}
+
+
```

Comment:
Question: Do we at any point make any assumptions about the column index order? It looks to me like we only use them to load the rhs, so I don't see how this could fail if the sorted one succeeds

Reply:
No, I don't think we do that, but I think it is always good to check if it still works.
Additionally, it is possible that we add another implementation where we do. In that case, it is more likely that we add an unsorted test for that as well since we see that it is tested for the others.

Reply:
I tend to disagree a bit there, since I can't imagine any benefit you gain from relying on the column order in an SpMV algorithm (which the cache hierarchy doesn't already provide you).

Anyways, the main question I wanted to ask here is: Do we need these changes in this PR, since they are unrelated to Jacobi and ILU? We don't test other parts of the implementation that might change depending on the column order either.

Reply:
I can definitively remove it. We don't need it as part of this PR, i just added it to see if we depend on the sorting somewhere else.